test/latest/10-canonicalize.js

```diff
@@ -0,0 +1,281 @@
+        in the headers Signature Parameter, in the order they appear
+        in the headers Signature Parameter.`, async function() {
+      /**
+        * In order to generate the string that is signed with a key, the client
+        * MUST use the values of each HTTP header field in the `headers`
+        * Signature Parameter, in the order they appear in the `headers`
+        * Signature Parameter.
+       */
+        generatorOptions.args.headers = ['Content-Length', 'Host'];
+        const result = await util.generate(
+          'default-test', generatorOptions);
+        expect(result, 'Expected a result').to.exist;
+        result.should.be.a('string');
+        let expected = 'content-length: 18\n';
+        expected += 'host: example.com\n';
+        result.should.equal(expected, 'expected signature string to match');
+      });
+
+      it(`All header field values associated with the header field
+        MUST be concatenated, separated by an ASCII comma and an ASCII space,
+        and used in the order in which they will appear in the
+        transmitted HTTP message.`, async function() {
+      /**
+        * If there are multiple instances of the same header
+        * field, all header field values associated with the
+        * header field MUST be concatenated, separated by an
+        * ASCII comma and an ASCII space `, `, and used in the
+        * order in which they will appear in the transmitted HTTP message.
+      */
+      });
+
+      it(`If a header specified in the headers parameter cannot be
+        matched with a provided header in the message, the implementation
+        MUST produce an error.`, async function() {
+      /**
+        * If a header specified in the headers parameter is
+        * malformed or cannot be matched with a provided
+        * header in the message, the implementation
+        * MUST produce an error.
+      */
+        let error = null;
+        try {
+          generatorOptions.args.headers = ['not-in-request'];
+          await util.generate('basic-request', generatorOptions);
+        } catch(e) {
+          error = e;
+        }
+        expect(error, 'Expected an error to be thrown').to.exist;
+      });
+      it(`If a header specified in the headers parameter is
+        malformed the implementation MUST produce an error.`, async function() {
+      /**
+        * If a header specified in the headers parameter is
+        * malformed or cannot be matched with a provided
+        * header in the message, the implementation
+        * MUST produce an error.
+      */
+        let error = null;
+        generatorOptions.args.headers = ['Server'];
+        try {
+          await util.generate(
+            'malformed-request', generatorOptions);
+        } catch(e) {
+          error = e;
+        }
+        expect(error, 'Expected an error to be thrown').to.exist;
+      });
+
+      it(`If the header value is a zero-length string,
```

Comment:
should this be it.skip or something to indicate that it's not a passing test?  Otherwise, it does show as passing correct?